cloudify_cli/utils.py

```diff
@@ -344,8 +347,21 @@ def get_rest_client(manager_ip=None, rest_port=None, protocol=None):
 
     trust_all = get_ssl_trust_all()
 
-    return CloudifyClient(host=manager_ip, port=rest_port, protocol=protocol,
-                          headers=headers, cert=cert, trust_all=trust_all)
+    client = CloudifyClient(host=manager_ip, port=rest_port, protocol=protocol,
+                            headers=headers, cert=cert, trust_all=trust_all)
+
+    if skip_version_check:
+        return client
+
+    cli_version, manager_version = get_cli_manager_versions()
+
+    if cli_version == manager_version:
+        return client
+    else:
+        message = ('CLI and manager versions do not match\n'
+                   'Cloudify CLI: {}\n'
+                   'Cloudify Manager: {}').format(cli_version, manager_version)
+        sys.exit(message)
```

Comment:
There isn't any particular reason. Should I use CloudifyCliError (I assume that you meant this exception)?